src/main/java/com/couchbase/lite/Database.java

```diff
@@ -2882,37 +2896,28 @@ public void insertAttachmentForSequenceWithNameAndType(InputStream contentStream
      */
     @InterfaceAudience.Private
     public void insertAttachmentForSequenceWithNameAndType(long sequence, String name, String contentType, int revpos, BlobKey key) throws CouchbaseLiteException {
-        try {
-            ContentValues args = new ContentValues();
-            args.put("sequence", sequence);
-            args.put("filename", name);
-            if (key != null) {
-                args.put("key", key.getBytes());
-                args.put("length", attachments.getSizeOfBlob(key));
-            }
-            args.put("type", contentType);
-            args.put("revpos", revpos);
-            long result = database.insert("attachments", null, args);
-            if (result == -1) {
-                String msg = "Insert attachment failed (returned -1)";
-                Log.e(Database.TAG, msg);
-                throw new CouchbaseLiteException(msg, Status.INTERNAL_SERVER_ERROR);
-            }
-        } catch (SQLException e) {
-            Log.e(Database.TAG, "Error inserting attachment", e);
-            throw new CouchbaseLiteException(e, Status.INTERNAL_SERVER_ERROR);
-        }
+        insertAttachmentForSequenceWithNameAndType(sequence, name, contentType, revpos, key, key != null ? attachments.getSizeOfBlob(key): -1, AttachmentInternal.AttachmentEncoding.AttachmentEncodingNone, -1);
     }
 
-    private void insertAttachmentForSequenceWithNameAndType(long sequence, String name, String contentType, int revpos, BlobKey key, long length) throws CouchbaseLiteException {
+    private void insertAttachmentForSequenceWithNameAndType(long sequence, String name, String contentType, int revpos, BlobKey key, long length, AttachmentInternal.AttachmentEncoding encoding, long encodedLength) throws CouchbaseLiteException {
         try {
             ContentValues args = new ContentValues();
             args.put("sequence", sequence);
             args.put("filename", name);
-            args.put("key", key.getBytes());
-            args.put("length", length);
             args.put("type", contentType);
             args.put("revpos", revpos);
+            if (key != null) {
+                args.put("key", key.getBytes());
+            }
+            if (length >= 0) {
+                args.put("length", length);
+            }
+            if(encoding == AttachmentInternal.AttachmentEncoding.AttachmentEncodingGZIP) {
+                args.put("encoding", encoding.ordinal());
+                if (encodedLength >= 0) {
```

Comment:
encodedLength is information only. In case encodedLength is less than 0, the code does not store the value. It will set null value in encoded-length column in SQLite attachments table.